src/TriggerBinding/SqlTriggerBinding.cs

```diff
@@ -0,0 +1,186 @@
+        /// <param name="value">
+        /// The list of <see cref="SqlChange{T}"/> data.
+        /// </param>
+        /// <param name="context">
+        /// Unused
+        /// </param>
+        /// <exception cref="InvalidOperationException">
+        /// Thrown if "value" is not of type IReadOnlyList<SqlChange<T>>.
+        /// </exception>
+        /// <returns>
+        /// The ITriggerData which stores the list of SQL table changes as well as the SimpleValueBinder
+        /// </returns>
+        public Task<ITriggerData> BindAsync(object value, ValueBindingContext context)
+        {
+            if (!(value is IReadOnlyList<SqlChange<T>> changes))
+            {
+                throw new InvalidOperationException("The value passed to the SqlTrigger BindAsync must be of type IReadOnlyList<SqlChange<T>>");
+            }
+
+            var bindingData = new Dictionary<string, object>(StringComparer.OrdinalIgnoreCase)
+            {
+                { "SqlTrigger", changes }
+            };
+
+            return Task.FromResult<ITriggerData>(new TriggerData(new SimpleValueProvider(this._parameter.ParameterType, changes, this._tableName), bindingData));
+        }
+
+        /// <summary>
+        /// Creates a listener that will monitor for changes to the user's table
+        /// </summary>
+        /// <param name="context">
+        /// Context for the listener, including the executor that executes the user's function when changes are detected in the user's table
+        /// </param>
+        /// <exception cref="ArgumentNullException">
+        /// Thrown if context is null
+        /// </exception>
+        /// <returns>
+        /// The listener
+        /// </returns>
+        public async Task<IListener> CreateListenerAsync(ListenerFactoryContext context)
+        {
+            if (context == null)
+            {
+                throw new ArgumentNullException(nameof(context), "Missing listener context");
+            }
+
+            string userFunctionId = await this.GetUserFunctionIdAsync();
+            return new SqlTriggerListener<T>(this._connectionString, this._tableName, userFunctionId, context.Executor, this._logger);
+        }
+
+        /// <returns> A description of the SqlTriggerParameter (<see cref="SqlTriggerParameterDescriptor"/> </returns>
+        public ParameterDescriptor ToParameterDescriptor()
+        {
+            return new SqlTriggerParameterDescriptor
+            {
+                Name = this._parameter.Name,
+                Type = "SqlTrigger",
+                TableName = _tableName
+            };
+        }
+
+        private async Task<string> GetUserFunctionIdAsync()
+        {
+            string hostId = await this._hostIdProvider.GetHostIdAsync(CancellationToken.None);
+
+            var methodInfo = (MethodInfo)this._parameter.Member;
+            string functionName = $"{methodInfo.DeclaringType.FullName}.{methodInfo.Name}";
+
+            using (var sha256 = SHA256.Create())
```

Comment:
What is this logic doing - did you copy it from somewhere?

Reply:
The logic is used to have a unique ID for identifying user function. This was inspired from TimerTrigger code
https://github.com/Azure/azure-webjobs-sdk-extensions/blob/01b1a826351bdd1883100d2af2470f1bde88ae5c/src/WebJobs.Extensions/Extensions/Timers/Bindings/TimerTriggerBinding.cs#L39-L40

Reply:
Added the explanation in the method summary.